core/src/main/scala/org/apache/spark/SparkHadoopWriter.scala

```diff
@@ -41,20 +41,20 @@ class SparkHadoopWriter(@transient jobConf: JobConf)
   with SparkHadoopMapRedUtil
   with Serializable {
 
-  private val now = new Date()
-  private val conf = new SerializableWritable(jobConf)
+  protected val now = new Date()
+  protected val conf = new SerializableWritable(jobConf)
 
-  private var jobID = 0
-  private var splitID = 0
-  private var attemptID = 0
-  private var jID: SerializableWritable[JobID] = null
-  private var taID: SerializableWritable[TaskAttemptID] = null
+  protected var jobID = 0
+  protected var splitID = 0
+  protected var attemptID = 0
+  protected var jID: SerializableWritable[JobID] = null
+  protected var taID: SerializableWritable[TaskAttemptID] = null
 
-  @transient private var writer: RecordWriter[AnyRef,AnyRef] = null
-  @transient private var format: OutputFormat[AnyRef,AnyRef] = null
-  @transient private var committer: OutputCommitter = null
-  @transient private var jobContext: JobContext = null
-  @transient private var taskContext: TaskAttemptContext = null
+  @transient protected var writer: RecordWriter[AnyRef,AnyRef] = null
+  @transient protected var format: OutputFormat[AnyRef,AnyRef] = null
+  @transient protected var committer: OutputCommitter = null
+  @transient protected var jobContext: JobContext = null
+  @transient protected var taskContext: TaskAttemptContext = null
```

Comment:
I changed the scope of var/def of SparkHadoopWriter to reuse these code in orc writting api implementation